compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala

```diff
@@ -29,7 +29,8 @@ final class Conversions(
 
   // The ledger data will not contain information from the partial transaction at this point.
   // We need the mapping for converting error message so we manually add it here.
-  private val ptxCoidToNodeId = incomplete.transaction.nodes
+  private val ptxCoidToNodeId = incomplete
+    .fold[Map[NodeId, N.GenNode[NodeId, V.ContractId]]](Map.empty)(_.transaction.nodes)
```

Comment:
```suggestion
  private val ptxCoidToNodeId = incomplete.map(_.transaction.nodes).getOrElse(Map.empty)
```
I prefer `fold` as well in general but the need to explicitly spell out the type could make `map`/`getOrElse` preferable here. Feel free to discard this suggestion.

Reply:
I blame Scala for having shitty type inference.